### ISSUE TITLE
Use HEAD request, rather than GET request

### ISSUE THREAD

**Issue by gep13** [gep13/chocolatey-vscode#123]
In this location:

https://github.com/gep13/chocolatey-vscode/blob/develop/src/Chocolatey.Language.Server/UriExtensions.cs#L17

This will mean that no response is required to be sent to the client, and should speed things up.

Should be possible using HttpWebRequest, which can set the HEAD method type:

https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.method?view=netframework-4.7.2

**Comment by mkevenaar** [comment]
@gep13 HTTP HEAD is not supported for all websites. Microsoft for example does not support the HEAD. That's why I have switched the validator to use full GET requests. 

Implementing this **will** result in false positives.

**Comment by gep13** [comment]
@mkevenaar when you say it is not supported, what do you get back from the website when you make the request?  Would it be possible to first make a HEAD request, and then knowingly attempt a GET request?

Or better, could we change it to make a HEAD request, and only attempt a GET request in the event of a failure?

**Comment by mkevenaar** [comment]
I believe Microsoft returned a HTTP status of 503.

Original URL: http://go.microsoft.com/fwlink/?LinkID=206977

```
curl -X HEAD -k -i https://www.microsoft.com/en-us/legal/intellectualproperty/copyright/default.aspx
HTTP/1.1 503 Service Unavailable
Mime-Version: 1.0
Content-Type: text/html
Content-Length: 177
Expires: Tue, 26 Feb 2019 20:47:11 GMT
X-Reference-Error: 119.8615655f.1551214031.48fd44e5
Date: Tue, 26 Feb 2019 20:47:11 GMT
Connection: keep-alive
Strict-Transport-Security: max-age=31536000
X-RTag: RT
```

I will look into a solution to implement this.

**Comment by mkevenaar** [comment]
fixed with 7b90b47

**Comment by gep13** [comment]
@mkevenaar thanks for closing this! 👍
